x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/persistence/TrainedModelProvider.java

```diff
@@ -381,14 +381,24 @@ public void deleteTrainedModel(String modelId, ActionListener<Boolean> listener)
     public void expandIds(String idExpression,
```

Comment:
Could you add a unit test to `TrainedModelProviderTests` for the new functionality?

Reply:
I did add a test verifying that filters are added to the expansion query.